iree/compiler/Dialect/Util/IR/UtilOps.td

```diff
@@ -176,6 +176,37 @@ def Util_RangeExtentsOp : Util_PureOp<"range.extents", [
   let hasCanonicalizer = 1;
 }
 
+//===----------------------------------------------------------------------===//
+// Alignment Arithmetic
+//===----------------------------------------------------------------------===//
+
+def Util_AlignOp : Util_PureOp<"align", [
+  AllTypesMatch<["value", "result"]>
+  ]> {
+  let summary = [{returns the (value + (alignment - 1)) & ~(alignment - 1)}];
```

Comment:
Summary should probably be English and not this math :P 

The math can go in the description if you think it really matters (though it is an implementation detail, so I'd say omit it)